core/components/Themes.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace luya\components;
+
+use Yii;
+use luya\Exception;
+
+/**
+ * Theme for LUYA CMS.
+ *
+ * This module / component allow user to manage actual display themes.
+ *
+ * @author Mateusz Szymański Teamwant <zzixxus@gmail.com>
+ * @author Mateusz Szymański Teamwant <kontakt@teamwant.pl>
+ * @since 1.0.0
+ */
+class Themes extends \yii\base\Component
+{
+
+    /**
+     * @var array
+     */
+    private $config = array();
+    /**
+     * @var array
+     */
+    private $themes = array();
+
+
+    /**
+     * Initialize Themes Component
+     */
+    public function init()
+    {
+        parent::init();
+        $this->setup();
+    }
+
+
+    /**
+     * Setup component
+     * @return $this
+     */
+    public function setup()
+    {
+
+        $this->config['active'] = $this->getActive();
+        $this->config['views'] = $this->config['active'] . '/views';
+        $this->config['resources'] = $this->config['active'] . '/resources';
+        $this->config['layout'] = $this->config['views'] . '/layouts';
+        $this->config['layout_file'] = 'main';
+        $this->config['layout_main'] = $this->config['layout'] . '/' . $this->config['layout_file'] . '.php';
+
+        if ($data = $this->getThemeData($this->config['active'])) {
+            $this->config['theme'] = $data;
+        } else {
+            $this->config['theme'] = ['error' => 1];
+        }
+
+
+        Yii::$app->params['active_theme'] = $this->config;
+
+
+        return $this;
+    }
+
+    /**
+     * Get current theme from SQL
+     * If you have own theme you must inset into db record named active_theme like that:
+     *
+     * INSERT INTO `admin_config` (`name`, `value`, `is_system`, `id`) VALUES ('active_theme', '@app/themes/teamwant', '1', NULL);
+     *
+     * todo: create admin module to manage themes
+     * @return string
+     * @throws \yii\db\Exception
+     */
+    public function getActive()
+    {
+        try {
```

Comment:
This won't work for other DBMS system therefore we would need active query, but its not allowed to make database requests in LUYA core as its mvc only and no database is needed. So the CMS must inject this information to the themes component in a bootstrap section for example.